nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TransformXml.java

```diff
@@ -372,4 +382,27 @@ private Source getSecureSource(final StreamSource streamSource) throws Transform
             throw new TransformerConfigurationException("XSLT Source Stream Reader creation failed", e);
         }
     }
+}
+
+class ErrorListenerLogger implements ErrorListener {
+    private final ComponentLog logger;
+
+    ErrorListenerLogger(ComponentLog logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void warning(TransformerException exception) throws TransformerException {
+        logger.warn(exception.getMessageAndLocation());
+    }
+
+    @Override
+    public void error(TransformerException exception) throws TransformerException {
+        logger.error(exception.getMessageAndLocation());
+    }
+
+    @Override
+    public void fatalError(TransformerException exception) throws TransformerException {
```

Comment:
```suggestion
    public void fatalError(TransformerException exception) {
```

Reply:
Isn't `throws TransformerException` part of the signature of the method interface?

Reply:
Yes, but if it isn't thrown, it doesn't need to be declared on the implementing method.

In this particular case, on further review, `fatalError` should throw the exception to prevent further processing.

Reply:
So I just rethrow the exception argument like below ?

`throw exception;`

Reply:
Yes, thanks!